src/CMakeLists.txt

```diff
@@ -246,17 +230,151 @@ ADD_DEFINITIONS(-DEIGEN_MPL2_ONLY)
 # - external if CERES_DIR_HINTS and FIND_PACKAGE return a valid Ceres setup
 # ==============================================================================
 FIND_PACKAGE(Ceres QUIET HINTS ${CERES_DIR_HINTS})
-IF(Ceres_FOUND)
-   MESSAGE("CERES FOUND")
-ELSE()
-  MESSAGE("CERES NOT FOUND")
+IF(NOT Ceres_FOUND)
   SET(OpenMVG_USE_INTERNAL_CERES ON)
   SET(CERES_INCLUDE_DIRS
     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/include
     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/internal/ceres/miniglog
     ${PROJECT_BINARY_DIR}/third_party/ceres-solver/config)
+    FILE(READ "${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/VERSION" CERES_CONFIG)
+    STRING(REGEX REPLACE "version ([0-9.]+).*" "\\1" CERES_VERSION ${CERES_CONFIG})
   SET(CERES_LIBRARIES ceres cxsparse)
-ENDIF(Ceres_FOUND)
+ENDIF()
+
+# ==============================================================================
+# Flann
+# ==============================================================================
+# - internal by default (flann),
+# - external if FLANN_DIR_HINTS and a valid Flann setup is found
+# ==============================================================================
+FIND_PATH(FLANN_INCLUDE_DIRS flann.hpp ${FLANN_DIR_HINTS})
```

Comment:
@simogasp , @pmoulon 

* There is no native find_package() for clp, coinutils, osi, lemon, flann.

* Yes LIBNAME_DIR_HINTS tell /hints where headers are. Here i could use additional PATH_SUFFIXES (that can lookup all standard system include paths too) but then i am afraid it will find _always_ find the header files, so better give it by _exact_ way or use internal one.

* For the ".so" FIND_LIBRARY() will found them by looking up all your standard  $LD_LIBRARY_PATH on the system. Yes he has "clues" so no need to specify an exact location for libs too. If libs are not in LD_LIBRARY_PATH than there is a terrible situation not only finding the libs but then using them by runtime.

-----<<<---

Now that these are more clear, let me further propose variants:

1. Leave it just how is it now with find_library() & find_path() tandem.
2. Do "Find_packageX.cmake" for each (will relay on proposal no1 indeed).
3. Use special cmake's FindPkgConfig & Pkg_Check_Modules(Clp clp).

-----<<<----

* For No.3 proposal Clp, CoinUtils, Osi and Lemon seems to have the 'pkg-config' style thing. 

$ rpm -ql coin-or-Clp-devel | grep .pc
/usr/lib64/pkgconfig/clp.pc
/usr/lib64/pkgconfig/osi-clp.pc

* Now, a lookup for such package would look like few lines:

// require pkg-config tool.
INCLUDE(FindPkgConfig REQUIRED)
// search Clp
IF(PKG_CONFIG_FOUND)
    INCLUDE(FindPkgConfig)
    PKG_CHECK_MODULES(Clp clp)
ENDIF()

* After these lines ${CLP_INCLUDE_DIRS} #{CLP_LIBRARIES} are ready to consume.

I'll vote No.3 if do not mine to bring additional pkg-config rool dependency here, and with this i believe we consume all ideas, at least on my side.

Thanks !